### PR TITLE
[react-navigation] Added StackRouter and TabRouter

### PR DESCRIPTION
**Pull request by iRoachie**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] TabRouter <<https://github.com/react-community/react-navigation/blob/master/src/routers/TabRouter.js#L25>> 
- [x] StackRouter <<https://github.com/react-community/react-navigation/blob/master/src/routers/StackRouter.js#L31>> 

TabRouter here - https://reactnavigation.org/docs/routers/tab
StackRouter here - https://reactnavigation.org/docs/routers/stack

### CODE REVIEW

**Comment by iRoachie** [comment]
@petejkim Everything good here?

**Comment by iRoachie** [comment]
@alloy

**Comment by alloy** [comment]
I’ll only merge if a maintainer of a set of types gives the thumbs-up, which I’m personally not on this project.

**Comment by iRoachie** [comment]
Ah okay, got you now

File: types/react-navigation/index.d.ts [hunk 1]
```diff
@@ -633,6 +634,19 @@ export class Transitioner extends React.Component<
...
+ * @desc from react-navigation/src/routers/TabRouter.js
+ * @param routeConfigs
+ * @param config
+ */
+export function TabRouter(
+  routeConfigs: NavigationRouteConfigMap,
+  config: NavigationTabRouterConfig
+): NavigationRouter<any, any, any>
+
 /**
```
Comment (huhuanming) [inline types/react-navigation/index.d.ts]: `StackRouter` is familar with `TabRouter`.
If it's convenient, would you add StackRouter too.
😄

Comment (iRoachie) [inline types/react-navigation/index.d.ts]: Lol sure 😅

Comment (iRoachie) [inline types/react-navigation/index.d.ts]: @huhuanming Updated to add StackRouter

**Comment by iRoachie** [comment]
@huhuanming Good now?

**Comment by iRoachie** [comment]
@fangpenlin @mhcgrq @petejkim

**Comment by huhuanming** [comment]
👍

**Comment by iRoachie** [comment]
@alloy "The magical thumbs have been granted"

**Comment by huhuanming** [comment]
😳

**Comment by iRoachie** [comment]
@huhuanming Can you check this one https://github.com/DefinitelyTyped/DefinitelyTyped/pull/17712

**Comment by huhuanming** [comment]
sure.

**Comment by alloy** [comment]
Alrighty 👍

**Pull request merged**